influxql/iterator.go

```diff
@@ -509,3 +585,24 @@ type nilFloatIterator struct{}
 
 func (*nilFloatIterator) Close() error      { return nil }
 func (*nilFloatIterator) Next() *FloatPoint { return nil }
+
+type castFloatIterator struct {
+	input IntegerIterator
+}
```

Comment:
Can you rename this to something like `integerFloatCastIterator` in case we add additional casting iterators to `Float` in the future?